eventmesh-connectors/eventmesh-connector-prometheus/src/main/java/org/apache/eventmesh/connector/prometheus/source/connector/PrometheusSourceConnector.java

```diff
@@ -0,0 +1,189 @@
+import org.apache.eventmesh.openconnect.api.source.Source;
+import org.apache.eventmesh.openconnect.offsetmgmt.api.data.ConnectRecord;
+import org.apache.eventmesh.openconnect.offsetmgmt.api.data.RecordOffset;
+import org.apache.eventmesh.openconnect.offsetmgmt.api.data.RecordPartition;
+import org.apache.eventmesh.openconnect.offsetmgmt.api.storage.OffsetStorageReader;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+
+@Slf4j
+public class PrometheusSourceConnector implements Source {
+
+    private static final String INSTANCE_ID = "instanceId";
+
+    private OffsetStorageReader offsetStorageReader;
+
+    private PrometheusSourceConfig sourceConfig;
+
+    private CloseableHttpClient httpClient;
+
+    private QueryPrometheusReq queryPrometheusReq;
+
+    private Long startTime;
+
+    private Long initTime;
+
+    private Integer interval;
+
+    private String url;
+
+    @Override
+    public Class<? extends Config> configClass() {
+        return PrometheusSourceConfig.class;
+    }
+
+    @Override
+    public void init(Config config) throws Exception {
+        this.sourceConfig = (PrometheusSourceConfig) config;
+        doInit();
+    }
+
+    @Override
+    public void init(ConnectorContext connectorContext) throws Exception {
+        SourceConnectorContext sourceConnectorContext = (SourceConnectorContext) connectorContext;
+        this.sourceConfig = (PrometheusSourceConfig) sourceConnectorContext.getSourceConfig();
+        this.offsetStorageReader = sourceConnectorContext.getOffsetStorageReader();
+
+        doInit();
+    }
+
+    private void doInit() {
+        queryPrometheusReq = new QueryPrometheusReq();
+        queryPrometheusReq.setQuery(sourceConfig.getConnectorConfig().getQuery());
+        queryPrometheusReq.setStep(sourceConfig.getConnectorConfig().getStep());
+
+        interval = sourceConfig.getConnectorConfig().getInterval();
+        initTime = sourceConfig.getConnectorConfig().getInitTime();
+
+        url = MessageFormat.format("{0}/{1}", sourceConfig.getConnectorConfig().getAddress(), sourceConfig.getConnectorConfig().getApi());
+
+        httpClient = HttpClientBuilder.create().build();
+    }
+
+    @Override
+    public void start() throws Exception {
+        log.info("prometheus source connector start.");
+
+        Map<String, String> partitionMap = new HashMap<>();
+        partitionMap.put(INSTANCE_ID, sourceConfig.getConnectorConfig().getConnectorId());
+        RecordPartition recordPartition = new RecordPartition(partitionMap);
+        RecordOffset recordOffset = offsetStorageReader.readOffset(recordPartition);
```

Comment:
why you use the recordOffset and offsetStorageReader for prometheus connector?

Reply:
In prometheus source connector ,I read data from prometheus server with time range, and use recordOffset to save completed timestamp for each query. Is that necessary?

Reply:
offsetmgmt is optional and recommended.

Reply:
> In prometheus source connector ,I read data from prometheus server with time range, and use recordOffset to save completed timestamp for each query. Is that necessary?

The purpose of reading and persisting offset is to ensure that message duplication is avoided after the source connector is restarted. Depending on how you use it here, it seems that it will cause ambiguity. I understand that promethues data synchronization should not require offset to avoid ambiguity.

---
offset 的读取与持久化是为了保证source connector重启后避免消息重复，看你这里用法感觉会造成歧义，我理解promethues的数据同步应该不需要offset，避免歧义

Reply:
OK let me check